internal/infoschema_test.go

```diff
@@ -153,4 +139,237 @@ func TestProcessInfoSchema(t *testing.T) {
+		{name: "string byte", srcType: schema.Type{Name: "bytea"}, spType: ddl.String{Len: ddl.MaxLength{}}, in: []byte("abc"), e: "abc"},
+		{name: "string int64", srcType: schema.Type{Name: "bigint"}, spType: ddl.String{Len: ddl.MaxLength{}}, in: int64(42), e: "42"},
+		{name: "string float64", srcType: schema.Type{Name: "float8"}, spType: ddl.String{Len: ddl.MaxLength{}}, in: float64(42.3), e: "42.3"},
+		{name: "string time", srcType: schema.Type{Name: "timestamp"}, spType: ddl.String{Len: ddl.MaxLength{}},
+			in: getTime(t, "2019-10-29T05:30:00+10:00"), e: "2019-10-29 05:30:00 +1000 +1000"},
+		{name: "timestamptz", srcType: schema.Type{Name: "timestamptz"}, spType: ddl.Timestamp{},
+			in: getTime(t, "2019-10-29T05:30:00+10:00"), e: getTime(t, "2019-10-29T05:30:00+10:00")},
+		{name: "timestamptz string", srcType: schema.Type{Name: "timestamptz"}, spType: ddl.Timestamp{},
+			in: "2019-10-29 05:30:00+10:00", e: getTime(t, "2019-10-29T05:30:00+10:00")},
+		{name: "timestamp", srcType: schema.Type{Name: "timestamptz"}, spType: ddl.Timestamp{},
+			in: getTime(t, "2019-10-29T05:30:00Z"), e: getTime(t, "2019-10-29T05:30:00Z")},
+		{name: "timestamp string", srcType: schema.Type{Name: "timestamptz"}, spType: ddl.Timestamp{},
+			in: "2019-10-29 05:30:00", e: getTime(t, "2019-10-29T05:30:00Z")},
+
+		// ConvertSqlRow uses convArray for conversion of array types.
+		// Since convArray is extensively tested in data_test.go, we
+		// only test a few cases here.
+		{name: "array bool", srcType: schema.Type{Name: "bool", ArrayBounds: []int64{-1}}, spType: ddl.Bool{}, isArray: true,
+			in: []byte("{true,false,NULL}"), e: []spanner.NullBool{
+				spanner.NullBool{Bool: true, Valid: true},
+				spanner.NullBool{Bool: false, Valid: true},
+				spanner.NullBool{Valid: false}}},
+		{name: "timestamp array", srcType: schema.Type{Name: "timestamptz", ArrayBounds: []int64{-1}}, spType: ddl.Timestamp{}, isArray: true,
+			in: []byte(`{"2019-10-29 05:30:00+10",NULL}`),
+			e: []spanner.NullTime{
+				spanner.NullTime{Time: getTime(t, "2019-10-29T05:30:00+10:00"), Valid: true},
+				spanner.NullTime{Valid: false}}},
+	}
+	tableName := "testtable"
+	for _, tc := range singleColTests {
+		col := "a"
+		conv := MakeConv()
+		conv.SetLocation(time.UTC)
+		cols := []string{col}
+		srcSchema := schema.Table{Name: tableName, ColNames: []string{col}, ColDefs: map[string]schema.Column{col: schema.Column{Type: tc.srcType}}}
+		spSchema := ddl.CreateTable{
+			Name:     tableName,
+			ColNames: []string{col},
+			ColDefs:  map[string]ddl.ColumnDef{col: ddl.ColumnDef{Name: col, T: tc.spType, IsArray: tc.isArray}}}
+		ac, av, err := ConvertSqlRow(conv, tableName, cols, srcSchema, tableName, cols, spSchema, []interface{}{tc.in})
+		assert.Equal(t, cols, ac)
+		assert.Equal(t, []interface{}{tc.e}, av)
+		assert.Nil(t, err)
+	}
+
+	// End-to-end test: builds schema from information_schema
```

Comment:
Nits: Should we split this into a separate test?

Reply:
Done.